core/src/main/java/com/linecorp/armeria/common/AggregatedHttpResponse.java

```diff
@@ -0,0 +1,247 @@
+        requireNonNull(trailingHeaders, "trailingHeaders");
+
+        final ResponseHeaders headers = ResponseHeaders.builder(status)
+                                                       .contentType(mediaType)
+                                                       .build();
+        return of(headers, content, trailingHeaders);
+    }
+
+    /**
+     * Creates a new HTTP response with empty content.
+     *
+     * @param headers the HTTP headers
+     */
+    static AggregatedHttpResponse of(ResponseHeaders headers) {
+        requireNonNull(headers, "headers");
+        return of(headers, HttpData.EMPTY_DATA, HttpHeaders.of());
+    }
+
+    /**
+     * Creates a new HTTP response.
+     *
+     * @param headers the HTTP headers
+     * @param content the content of the HTTP response
+     */
+    static AggregatedHttpResponse of(ResponseHeaders headers, HttpData content) {
+        requireNonNull(headers, "headers");
+        requireNonNull(content, "content");
+        return of(headers, content, HttpHeaders.of());
+    }
+
+    /**
+     * Creates a new HTTP response.
+     *
+     * @param headers the HTTP headers
+     * @param content the content of the HTTP response
+     * @param trailingHeaders the trailing HTTP headers
+     */
+    static AggregatedHttpResponse of(ResponseHeaders headers, HttpData content, HttpHeaders trailingHeaders) {
+        requireNonNull(headers, "headers");
+        requireNonNull(content, "content");
+        requireNonNull(trailingHeaders, "trailingHeaders");
+        return of(Collections.emptyList(), headers, content, trailingHeaders);
+    }
+
+    /**
+     * Creates a new HTTP response.
+     *
+     * @param informationals the informational class (1xx) HTTP headers
+     * @param headers the HTTP headers
+     * @param content the content of the HTTP response
+     * @param trailingHeaders the trailing HTTP headers
+     */
+    static AggregatedHttpResponse of(Iterable<ResponseHeaders> informationals, ResponseHeaders headers,
+                                     HttpData content, HttpHeaders trailingHeaders) {
+
+        requireNonNull(informationals, "informationals");
+        requireNonNull(headers, "headers");
+        requireNonNull(content, "content");
+        requireNonNull(trailingHeaders, "trailingHeaders");
+
+        // Set the 'content-length' header if possible.
+        final ResponseHeaders newHeaders = setOrRemoveContentLength(ResponseHeaders.of(headers),
+                                                                    content, trailingHeaders);
+
+        return new DefaultAggregatedHttpResponse(ImmutableList.copyOf(informationals),
+                                                 newHeaders, content, trailingHeaders);
+    }
+
+    /**
+     * Returns the {@link ResponseHeaders}.
+     */
+    @Override
+    ResponseHeaders headers();
+
+    /**
+     * Returns the informational class (1xx) HTTP headers.
+     */
+    List<ResponseHeaders> informationals();
```

Comment:
I guess the old API may have been like this too, but wondering is it better to separate informational like this instead of aggregating it all into a single `ResponseHeaders`?

Reply:
informational headers could have a status so we still need two separate fields for that?

Reply:
Ah cool then that makes sense.